app/controllers/vm_common.rb

```diff
@@ -990,7 +916,7 @@ def set_checked_items
     if params[:all_checked]
       ids = params[:all_checked].split(',')
       ids.each do |id|
-        id = id.split('-')[1]
+        id = id == 'root' ? session[:genealogy_tree_root_id] : from_cid(parse_nodetype_and_id(id).last)
```

Comment:
348: +    base = params[:id] == 'root' ? session[:genealogy_tree_root_id] : params[:id].split('-').last
919: +        id = id == 'root' ? session[:genealogy_tree_root_id] : from_cid(parse_nodetype_and_id(id).last)

Sounds like this could be `TreeBuilderGenealogy.foobar(params[:id])` (or `id`).. ;)